src/types.ts

```diff
@@ -130,3 +134,5 @@ export type FetchRequest = RequestInfo;
 export interface SearchParameters {
   [key: string]: any;
 }
+
+type Arrayable<T> = T | Array<T>;
```

Comment:
I'm not sure about this one. It seems simpler as interceptor types are only defined once, but the more verbose approach might also be viable.

```ts
onRequest?: 
  | ((context: FetchContext) => Promise<void> | void)
  | Array<(context: FetchContext) => Promise<void> | void>;
```